libs/agent-sm/agent-policy/src/main/java/org/opensearch/secure_sm/policy/GrantNode.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.secure_sm.policy;
+
+import java.io.PrintWriter;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.LinkedList;
+
+public class GrantNode {
```

Comment:
I don't like the use of the word Node in here or PermissionNode. Node in OpenSearch refers to an instance of OpenSearch running in a cluster and outside of OpenSearch its generally used in the context of graphs. wdyt of simply calling this `Grant` or `GrantBlock` and have a `Grant` be composed of `PermissionRecord` or `PermissionElement`?

Reply:
The reference comes from the parser "Recursive Descent Parser" which is used to parse the grammer where each nodes representing elements of the grammar being parsed. 

But I am fine to rename them as plain ```GrantElement``` & ```PermissionElement```